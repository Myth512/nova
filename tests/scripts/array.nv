arr1 := [1, 2, 3]
arr2 := [4, 5, 6]
arr3 := [1, 2, 3]
arr4 := [3, 2, 1]

print(arr1) // Case 1: [1, 2, 3]
print(arr2) // Case 2: [4, 5, 6]

arr1.push(4)
print(arr1) // Case 3: [1, 2, 3, 4]

arr1.pop()
print(arr1) // Case 4: [1, 2, 3]

arr1.insert(1, 10)
print(arr1) // Case 5: [1, 10, 2, 3]

arr1.reverse()
print(arr1) // Case 6: [3, 2, 10, 1]

arr5 := arr1 + arr2
print(arr5) // Case 7: [3, 2, 10, 1, 4, 5, 6]

arr6 := arr1 * 2
print(arr6) // Case 8: [3, 2, 10, 1, 3, 2, 10, 1]

arr1[1] = 20
print(arr1) // Case 9: [3, 20, 10, 1]


print(arr1[2]) // Case 10: 10

print(len(arr1)) // Case 11: 4

print(arr1 == arr3) // Case 12: false
print(arr1 == arr4) // Case 13: false
print(arr1 == [3, 20, 10, 1]) // Case 14: true

print(arr1 != arr3) // Case 15: true
print(arr1 != arr4) // Case 16: true
print(arr1 != [3, 20, 10, 1]) // Case 17: false

print(arr1 > arr2) // Case 18: false
print(arr2 > arr1) // Case 19: true

print(arr1 >= arr3) // Case 20: true
print(arr1 >= arr2) // Case 21: false

print(arr1 < arr2) // Case 22: true
print(arr2 < arr1) // Case 23: false

print(arr3 <= arr1) // Case 24: true
print(arr2 <= arr1) // Case 25: false

arr1 += [15]
print(arr1) // Case 26: [3, 20, 10, 1, 15]

arr7 := [8, 3.14]
arr7 *= 3
print(arr7)
