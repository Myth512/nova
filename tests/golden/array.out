[1, 2, 3]
[4, 5, 6]
[1, 2, 3, 4]
[1, 2, 3]
[1, 10, 2, 3]
[3, 2, 10, 1]
[3, 2, 10, 1, 4, 5, 6]
[3, 2, 10, 1, 3, 2, 10, 1]
[3, 20, 10, 1]
10
4
false
false
true
true
true
false
false
true
true
false
true
false
true
false
[3, 20, 10, 1, 15]
[8, 3.14, 8, 3.14, 8, 3.14]
